aptos-move/framework/aptos-framework/sources/fungible_asset.move

```diff
@@ -434,20 +453,53 @@ module aptos_framework::fungible_asset {
             ref.metadata == store_metadata(store),
             error::invalid_argument(ETRANSFER_REF_AND_STORE_MISMATCH),
         );
+        set_frozen_flag_internal(store, frozen)
+    }
+
+    public(friend) fun set_frozen_flag_internal<T: key>(
+        store: Object<T>,
+        frozen: bool
+    ) acquires FungibleStore {
         let store_addr = object::object_address(&store);
         borrow_global_mut<FungibleStore>(store_addr).frozen = frozen;
 
         event::emit(Frozen { store: store_addr, frozen });
     }
 
+    /// Used by coin.move for the migration from coin to fungible asset only
+    public(friend) fun get_mint_ref_internal(metadata: Object<Metadata>): MintRef {
+        MintRef { metadata }
+    }
+
+    /// Used by coin.move for the migration from coin to fungible asset only
+    public(friend) fun get_transfer_ref_internal(metadata: Object<Metadata>): TransferRef {
+        TransferRef { metadata }
+    }
+
+    /// Used by coin.move for the migration from coin to fungible asset only
+    public(friend) fun get_burn_ref_internal(metadata: Object<Metadata>): BurnRef {
+        BurnRef { metadata }
+    }
+
```

Comment:
do we want to make these out of thin air or just store them in the table? not sure from an auditing perspective but this feels a little more risky.